packages/core/lib/debug/printer.js

```diff
@@ -171,6 +171,22 @@ class DebugPrinter {
     const colorizedLines = splitLines(colorizedSource);
 
     this.config.logger.log("");
+    // We print a warning message and display the end of source code when the
+    // instruction's byte-offset to the start of the range in the source code
+    // is past the end of source code
+    const instruction = this.session.view(solidity.current.instruction);
+    if (instruction.start >= source.length) {
+      this.config.logger.log(
+        `${colors.bold("Warning:")} Past end of source, displaying end.`
+      );
+      this.config.logger.log("");
+      // We set the source range to the end of source code for printing purpose only
+      // Note that the object "lines" is the split lines of source code as defined above
+      range.lines = {
```

Comment:
I think you should avoid mutating `range.lines` here, seeing as `range` is a field on an argument that was passed in.  While I don't think it should cause a problem in the cases that will actually come up, to avoid potential problems in the future you should avoid making this modification... this should be a pure function and not mutate its inputs.